eZ/Publish/API/Repository/Tests/Values/User/Limitation/LanguageLimitationTest.php

```diff
@@ -490,6 +493,164 @@ public function testPublishVersionTranslationWhenUserHasNoAccessToAllLanguages()
+        $permissionResolver = $repository->getPermissionResolver();
+
+        $draft = $this->createMultilingualFolderDraft($contentService);
+        $content = $contentService->publishVersion($draft->getVersionInfo());
+
+        $limitationLanguages = [self::GER_DE];
+
+        // Validate that limitation value doesn't contain all version's languages
+        self::assertNotEmpty(array_diff($content->getVersionInfo()->languageCodes, $limitationLanguages));
+
+        $permissionResolver->setCurrentUserReference(
+            $this->createEditorUserWithLanguageLimitation($limitationLanguages)
+        );
+
+        return $content;
+    }
+
+    public function testCopyMoveHideAndUnhideContentWithLanguageLimitationsContainingAllTranslations(): void
+    {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $locationService = $repository->getLocationService();
+        $permissionResolver = $repository->getPermissionResolver();
+
+        $draft = $this->createMultilingualFolderDraft($contentService);
+        $content = $contentService->publishVersion($draft->getVersionInfo());
+
+        $limitationLanguages = [self::GER_DE, self::ENG_US, self::ENG_GB];
+
+        // Validate that limitation value contains all version's languages
+        self::assertEmpty(array_diff($content->getVersionInfo()->languageCodes, $limitationLanguages));
+
+        $permissionResolver->setCurrentUserReference(
+            $this->createEditorUserWithLanguageLimitation($limitationLanguages)
+        );
+
+        // Copy
+        $locationCreateStruct = new LocationCreateStruct(['parentLocationId' => 2]);
+        $clonedContent = $contentService->copyContent($content->contentInfo, $locationCreateStruct);
+
+        self::assertSame($content->getVersionInfo()->languageCodes, $clonedContent->getVersionInfo()->languageCodes);
+
+        // Move
+        $targetContent = $this->createMultilingualFolderDraft($contentService);
+        $contentLocation = $locationService->loadLocation($content->contentInfo->mainLocationId);
+        $targetLocation = $locationService->loadLocation($targetContent->contentInfo->mainLocationId);
+
+        $locationService->moveSubtree($contentLocation, $targetLocation);
+
+        self::assertSame($content->getVersionInfo()->languageCodes, $targetContent->getVersionInfo()->languageCodes);
+
+        // Hide
+        $hiddenLocation = $locationService->hideLocation($targetLocation);
+        $hiddenLocationContent = $hiddenLocation->getContent();
+        self::assertSame($content->getVersionInfo()->languageCodes, $hiddenLocationContent->getVersionInfo()->languageCodes);
+
+        // Unhide
+        $revealedLocation = $locationService->unhideLocation($hiddenLocation);
+        $revealedLocationContent = $revealedLocation->getContent();
+        self::assertSame($content->getVersionInfo()->languageCodes, $revealedLocationContent->getVersionInfo()->languageCodes);
```

Comment:
Please split this into separate tests. In this case it's justified to use `@depends`